### DISCUSSION TITLE
Minimum number of structures to label

### DISCUSSION THREAD

**Discussion by omidshy** [deepmodeling/dpgen#908] [Q&A] [answered]
Hi,

based on my understanding from the provided description for the `fp_task_min` key in `PARAM` file, I was expecting that the `dpgen` continues with the labeling (`02.fp`) step only if the number of candidate structures from the exploration (`01.model_devi`) step are larger than the value of the `fp_task_min`.
However, what actually happens is that the `dpgen` continues to the `02.fp` step even if the number of candidates are smaller than the `fp_task_min` value, but stops at the the end of the current iteration (i.e. after labeling the candidates).

Is this an expected behavior from `dpgen`?
If so, doesn't it make more sense to stop the run after `01.model_devi`, when the number of candidates are smaller than the `fp_task_min` value?

**Comment by HuangJiameng** [comment]
> I was expecting that the dpgen continues with the labeling (02.fp) step only if the number of candidate structures from the exploration (01.model_devi) step are larger than the value of the fp_task_min.

You can refer to https://github.com/deepmodeling/dpgen/blob/master/dpgen/generator/run.py#L312-L314
Could you provide more information to help locate the problem?

**Comment by omidshy** [comment]
Parameters for the labeling step in my `param.json` file:
```
"fp_style": "cp2k",
"shuffle_poscar": false,
"fp_task_max": 900,
"fp_task_min": 200,
"fp_pp_path": "./cp2k_basis_pot_files/",
"fp_pp_files": ["BASIS_MOLOPT", "GTH_POTENTIALS", "dftd3.dat"],
"external_input_path": "./cp2k_inp_files/template.inp",
```
Here is the content of the `dpgen.log` file:
```
14:32:22,078 - INFO : start running
14:32:22,101 - INFO : =============================iter.000000==============================
14:32:22,101 - INFO : -------------------------iter.000000 task 00--------------------------
14:32:22,147 - INFO : -------------------------iter.000000 task 01--------------------------
03:50:06,313 - INFO : -------------------------iter.000000 task 02--------------------------
03:50:06,387 - INFO : -------------------------iter.000000 task 03--------------------------
03:50:09,967 - INFO : -------------------------iter.000000 task 04--------------------------
04:14:04,081 - INFO : -------------------------iter.000000 task 05--------------------------
04:14:04,106 - INFO : -------------------------iter.000000 task 06--------------------------
04:14:05,145 - INFO : system 000 candidate :      4 in   5010   0.08 %
04:14:05,146 - INFO : system 000 failed    :     10 in   5010   0.20 %
04:14:05,146 - INFO : system 000 accurate  :   4996 in   5010  99.72 %
04:14:05,158 - INFO : system 000 accurate_ratio:   0.9972    thresholds: 1.0000 and 1.0000   eff. task min and max   -1  900   number of fp tasks:      4
04:14:05,623 - INFO : -------------------------iter.000000 task 07--------------------------
05:28:54,147 - INFO : -------------------------iter.000000 task 08--------------------------
05:28:54,237 - INFO : failed tasks:      0 in      4    0.00 % 
05:28:55,133 - INFO : failed frame:      0 in      4    0.00 % 
05:28:55,145 - INFO : =============================iter.000001==============================
05:28:55,145 - INFO : -------------------------iter.000001 task 00--------------------------
05:28:55,197 - INFO : -------------------------iter.000001 task 01--------------------------
05:28:55,210 - INFO : -------------------------iter.000001 task 02--------------------------
05:28:55,223 - INFO : -------------------------iter.000001 task 03--------------------------
05:28:55,223 - INFO : finished
```
based on my understanding from the `dpgen.log` file; the value of the `fp_task_min` from the `param.json` file is apparently ignored by `dpgen (0.10.6)`.

**Reply by poyi81** [reply]
Dear @omidshy ,
I am wondering how did you set up your training so that the `iter.000000` would have 99.72% accuracy. If you can provide your `param.json` file I would be appreciated. 
Thanks.

**Comment by HuangJiameng** [comment, marked as answer]
The value of the `fp_task_min` from the `param.json` file only works in the `make_train` step. In `fp` step, `fp_task_min` is '-1' in default. You can set it in `model_devi_jobs` part of the `param.json` file. See https://github.com/deepmodeling/dpgen/blob/master/dpgen/generator/run.py#L2465-L2471
